iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1811,16 +1808,28 @@ private List<String> getPartitions(DDLOperationContext context, Configuration jo
         String[] row =
             fetcher.convert(currSerDe.deserialize(value), currSerDe.getObjectInspector())
                 .toString().split("\t");
-        parts.add(HiveTableUtil.getParseData(row[PART_IDX], row[SPEC_IDX], mapper, tbl.spec().specId()));
+        parts.add(HiveTableUtil.getParseData(row[IcebergTableUtil.PART_IDX], row[IcebergTableUtil.SPEC_IDX],
+            mapper, tbl.spec().specId()));
       }
     }
     Collections.sort(parts);
     return parts;
   }
 
   @Override
-  public void validatePartSpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable, Map<String, String> partitionSpec)
-      throws SemanticException {
+  public void validatePartSpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable, Map<String, String> partitionSpec,
+      Configuration config) throws SemanticException {
```

Comment:
conf

Reply:
not relevant anymore, passing RewritePolicy now as you suggested.